python/lsst/daf/butler/script/exportCalibs.py

```diff
@@ -0,0 +1,147 @@
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from astropy.table import Table
+import logging
+import os
+from .. import Butler, CollectionType
+
+
+log = logging.getLogger(__name__)
+
+
+def parseCalibrationCollection(registry, collection, datasetTypes):
+    """Search a calibration collection for calibration datasets.
+
+    Parameters
+    ----------
+    registry : `lsst.daf.butler.Registry`
+        Butler registry to use.
+    collection : `str`
+        Collection to search.  This should be a CALIBRATION
+        collection.
+
+    Returns
+    -------
+    exportCollections : `list` [`str`]
+        List of collections to save on export.
+    exportDatasets : `list` [`lsst.daf.butler.queries.DatasetQueryResults`]
+        Datasets to save on export.
+    """
+    exportCollections = []
+    exportDatasets = []
+    for calibType in datasetTypes:
+        associations = registry.queryDatasetAssociations(calibType, collections=[collection])
+        for result in associations:
+            exportDatasets.append(result.ref)
+            exportCollections.append(result.ref.run)
+    return exportCollections, exportDatasets
+
+
+def exportCalibs(repo, directory, collections):
+    """Certify a set of calibrations with a validity range.
+
+    Parameters
+    ----------
+    repo : `str`
+        URI to the location of the repo or URI to a config file
+        describing the repo and its location.
+    directory : `str`
+        URI string of the directory to write the exported
+        calibrations.
+    collections : `list` [`str`]
+       Data collections to pull calibrations from.  Must be an
+       existing `~CollectionType.CHAINED` or
+       `~CollectionType.CALIBRATION` collection.
+
+    Returns
+    -------
+    datasetTable : `astropy.table.Table`
+        A table containing relevant information about the calibrations
+        exported.
+
+    Raises
+    ------
+    RuntimeError :
+        Raised if the output directory already exists.
+    """
+    butler = Butler(repo, writeable=False)
+
+    calibTypes = ["bias", "dark", "flat", "defects", "linearity", "bfk", "ptc", "crosstalk"]
```

Comment:
I'm a bit worried about this list being hard-coded. Do you have to patch this every time a new dataset type is added? What happens if you only want to export just the bias? Why can't you query for all dataset types that have `isCalibration=True` ?